### PR TITLE
[ONNX] Update ONNX doc for indexing export

### PR DESCRIPTION
**Pull request by BowenBao**
Adding example code for supported cases.

### CODE REVIEW

**Review by malfet — APPROVED** [review]

File: docs/source/onnx.rst [hunk 1]
```diff
@@ -249,6 +249,133 @@ E.g.: ::
...
+  # Slice indices
+  data[:3]
+
+  # Tensor indices
+  data[torch.tensor([[1, 2], [2, 3]])]
+  data[torch.tensor([2, 3]), torch.tensor([1, 2])]
+  data[torch.tensor([[1, 2], [2, 3]]), torch.tensor([2, 3])]
+  data[torch.tensor([2, 3]), :, torch.tensor([1, 2])]
+
+  # Ellipsis (not supported in scripting)
```
Comment (spandantiwari) [inline docs/source/onnx.rst:299]: Can we move this to the end, after boolean mask, and highlight "not supported in scripting" part. Also, we should describe in a sentence or two what it means not be supported in scripting but otherwise supported.

File: docs/source/onnx.rst [hunk 2]
```diff
@@ -249,6 +249,133 @@ E.g.: ::
...
+  # Multiple tensor indices if any has rank >= 2
+  data[torch.tensor([[1, 2], [2, 3]]), torch.tensor([2, 3])] = new_data
+
+  # Multiple tensor indices that are not consecutive
+  data[torch.tensor([2, 3]), :, torch.tensor([1, 2])] = new_data
+
+  # Tensor indices that includes negative values.
+  data[torch.tensor([1, -2]), torch.tensor([-2, 3])] = new_data
+
+If you are experiencing issues exporting indexing that belongs to the above supported patterns, please double check that
```
Comment (spandantiwari) [inline docs/source/onnx.rst:377]: THis is a useful comment. 

I was wondering if we can put something like this - `please double check that
you are exporting with the latest opset (opset_version=12).`  - at the very top of this section also, as a general guideline. That is use the latest opset as much as possible so as to get maximum support for indexing.

**Review by spandantiwari — APPROVED** [review]
LGTM.

File: docs/source/onnx.rst (hunk 1 above)
Comment (BowenBao) [reply docs/source/onnx.rst]: added explanation for scripting. Couldn't move this though, because the next line is an example stating supporting all of the above combined, and ellipsis is one of them. There is mask indexing left out at the end, because it cannot be used with other patterns. I added a line explicitly stating that out as well.
